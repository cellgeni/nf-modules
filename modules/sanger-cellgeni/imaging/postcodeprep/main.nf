process IMAGING_POSTCODEPREP {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "quay.io/cellgeni/postcode:0.2.0"

    input:
    tuple val(meta), path(profile), path(tabular_codebook), path(readout_file), val(R)

    output:
    tuple val(meta), path(reformatted_profile), path(starfish_codebook), path(barcode_0123_str), emit: for_decoding
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def readout = readout_file ? "--readouts ${readout_file}" : ""
    reformatted_profile = "${prefix}_reformatted_profile.npy"
    starfish_codebook = "${prefix}_starfish_codebook.json"
    barcode_0123_str = "${prefix}_barcodes_0123_str.txt"
    """
    postcodeprep.py \\
        --profile ${profile} \\
        --tabular_codebook ${reformatted_profile} \\
        --out_starfish_codebook ${starfish_codebook} \\
        --out_reformatted_profile ${reformatted_profile} \\
        --out_barcodes_0123_str ${barcode_0123_str} \\
        --R ${R} \\
        ${readout} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        postcodeprep: \$(postcodeprep.py --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    reformatted_profile = "${prefix}_reformatted_profile.npy"
    starfish_codebook = "${prefix}_starfish_codebook.json"
    barcode_0123_str = "${prefix}_barcodes_0123_str.txt"
    """
    echo ${args}
    
    touch ${reformatted_profile}
    touch ${starfish_codebook}
    touch ${barcode_0123_str}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        postcodeprep: \$(postcodeprep.py --version)
    END_VERSIONS
    """
}
