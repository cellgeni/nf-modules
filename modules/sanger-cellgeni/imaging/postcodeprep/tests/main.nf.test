// nf-core modules test imaging/postcodeprep
nextflow_process {

    name "Test Process IMAGING_POSTCODEPREP"
    script "../main.nf"
    process "IMAGING_POSTCODEPREP"

    tag "modules"
    tag "modules_sangercellgeni"
    tag "imaging"
    tag "imaging/postcodeprep"

    test("ISS - npy") {

        when {
            params.postcodeprep_args = "--coding_col_prefix Readouts* --codebook_targer_col gene "
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/ISS/decoding/merfish_transcript_profile.npy', checkIfExists: true),
                    file(params.modules_testdata_base_path + '/ISS/decoding/codebook.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + '/ISS/decoding/readouts.csv', checkIfExists: true),
                    6,
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("ISS - npy - stub") {

        options "-stub"

        when {
            params.postcodeprep_args = "--coding_col_prefix Readouts* --codebook_targer_col gene "
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/ISS/decoding/merfish_transcript_profile.npy', checkIfExists: true),
                    file(params.modules_testdata_base_path + '/ISS/decoding/codebook.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + '/ISS/decoding/readouts.csv', checkIfExists: true),
                    6,
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
