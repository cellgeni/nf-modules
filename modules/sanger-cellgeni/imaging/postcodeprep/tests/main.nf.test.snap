{
    "ISS - npy": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "iss"
                        },
                        "iss_reformatted_profile.npy:md5,8e33738958891bc39bd1693cc23974c6",
                        "iss_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "iss_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ],
                "for_decoding": [
                    [
                        {
                            "id": "iss"
                        },
                        "iss_reformatted_profile.npy:md5,8e33738958891bc39bd1693cc23974c6",
                        "iss_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "iss_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-28T15:03:15.927270522"
    },
    "merfish - npy - 1": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merfish_1"
                        },
                        "merfish_1_reformatted_profile.npy:md5,1767337178874628135d325468822e8a",
                        "merfish_1_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_1_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ],
                "for_decoding": [
                    [
                        {
                            "id": "merfish_1"
                        },
                        "merfish_1_reformatted_profile.npy:md5,1767337178874628135d325468822e8a",
                        "merfish_1_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_1_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-11T23:18:05.16106763"
    },
    "merfish - npy": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merfish"
                        },
                        "merfish_reformatted_profile.npy:md5,ee3dc5826c11d393aaf813b08026c5c2",
                        "merfish_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ],
                "for_decoding": [
                    [
                        {
                            "id": "merfish"
                        },
                        "merfish_reformatted_profile.npy:md5,ee3dc5826c11d393aaf813b08026c5c2",
                        "merfish_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-11T14:50:35.173031726"
    },
    "merfish - npy - 100000": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merfish_100000"
                        },
                        "merfish_100000_reformatted_profile.npy:md5,45f2e67209caf00d01113dc3802c8519",
                        "merfish_100000_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_100000_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ],
                "for_decoding": [
                    [
                        {
                            "id": "merfish_100000"
                        },
                        "merfish_100000_reformatted_profile.npy:md5,45f2e67209caf00d01113dc3802c8519",
                        "merfish_100000_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_100000_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-11T23:18:38.814057637"
    },
    "merfish - npy - 5000": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merfish_5000"
                        },
                        "merfish_5000_reformatted_profile.npy:md5,ee3dc5826c11d393aaf813b08026c5c2",
                        "merfish_5000_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_5000_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ],
                "for_decoding": [
                    [
                        {
                            "id": "merfish_5000"
                        },
                        "merfish_5000_reformatted_profile.npy:md5,ee3dc5826c11d393aaf813b08026c5c2",
                        "merfish_5000_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_5000_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-11T23:18:27.430999861"
    },
    "ISS - npy - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_reformatted_profile.npy:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test_starfish_codebook.json:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test_barcodes_0123_str.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ],
                "for_decoding": [
                    [
                        {
                            "id": "test"
                        },
                        "test_reformatted_profile.npy:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test_starfish_codebook.json:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test_barcodes_0123_str.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-11T14:40:00.276319431"
    },
    "merfish - npy - 200": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merfish_200"
                        },
                        "merfish_200_reformatted_profile.npy:md5,5a3d1b515873ce5fb3a85cde34c1bad2",
                        "merfish_200_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_200_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ],
                "for_decoding": [
                    [
                        {
                            "id": "merfish_200"
                        },
                        "merfish_200_reformatted_profile.npy:md5,5a3d1b515873ce5fb3a85cde34c1bad2",
                        "merfish_200_starfish_codebook.json:md5,846c498cb0b5363e5fcfa44e94bf60b7",
                        "merfish_200_barcodes_0123_str.txt:md5,1240bdd581957bc56a810734a5b39b0e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5b96538fa4f568da2baa52f59a47ee67"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-11T23:18:16.308555074"
    }
}